src/main/java/com/monumental/services/MonumentService.java

```diff
@@ -185,15 +198,29 @@ public void getRelatedTags(List<Monument> monuments) {
                 root.get("id").in(ids)
         );
 
-        List<Monument> monumentsWithTags = this.getWithCriteriaQuery(query);
+        List<Monument> monumentsWithRecords = this.getWithCriteriaQuery(query);
+
+        String capitalizedFieldName = fieldName.substring(0, 1).toUpperCase() + fieldName.substring(1);
 
-        Map<Integer, List<Tag>> tagsMap = new HashMap<>();
-        for (Monument monument : monumentsWithTags) {
-            tagsMap.put(monument.getId(), monument.getTags());
+        Map<Integer, List> map = new HashMap<>();
+        for (Monument monument : monumentsWithRecords) {
+            try {
+                map.put(monument.getId(), (List) Monument.class.getDeclaredMethod("get" + capitalizedFieldName).invoke(monument));
```

Comment:
woof

Reply:
I've gotten _so_ used to be able to do `monument[fieldName]` in Javascript, it pains me to write this kind of reflection